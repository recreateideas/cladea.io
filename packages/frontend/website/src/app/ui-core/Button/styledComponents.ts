import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
