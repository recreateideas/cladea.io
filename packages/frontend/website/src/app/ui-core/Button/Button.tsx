import { Link } from 'react-router-dom';
import { ButtonProps } from '@mui/material';
import { StyledButton } from './styledComponents';

interface IProps extends ButtonProps {
    withLink?: string;
}

export const Button = (props: IProps) => {
    const { withLink, ...buttonProps } = props;
    const button = <StyledButton {...buttonProps} />;
    return withLink ? <Link to={withLink}>{button}</Link> : button;
};

Button.displayName = 'Button';
