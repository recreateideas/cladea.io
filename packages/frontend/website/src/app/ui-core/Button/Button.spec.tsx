import { Button } from './Button';
import { TestProvider } from '../../../tests';

const { mountWithProviderAndTheme, fullMount } = TestProvider({});

describe('<Button/>', () => {
    it('should render', () => {
        const root = mountWithProviderAndTheme(<Button />);

        expect(root).toMatchSnapshot();
    });
    it('should render the Link if withLink is not undefined', () => {
        const root = fullMount(<Button {...{ withLink: 'http://bla-bla.com' }} />);
        const link = root.find('Link');
        expect(link.length).toBe(1);
    });
});
