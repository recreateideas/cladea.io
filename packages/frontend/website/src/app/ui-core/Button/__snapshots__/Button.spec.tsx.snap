// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render 1`] = `
Array [
  .c3 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  -webkit-letter-spacing: 0.02857em;
  -moz-letter-spacing: 0.02857em;
  -ms-letter-spacing: 0.02857em;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #12D8F7;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(18,216,247,0.04);
}

.c1.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c2.MuiButton-containedPrimary .MuiButton-label {
  color: #ffffff;
}

@media print {
  .c0 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c1:hover {
    background-color: transparent;
  }
}

<button
    className="c0 MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root c1 c2"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="c3 MuiTouchRipple-root"
    />
  </button>,
  "",
]
`;
